Sources/PostgresNIO/Data/PostgresData+Range.swift

```diff
@@ -0,0 +1,244 @@
+import NIOCore
+
+// MARK: PostgresRangeBound
+
+public protocol PostgresRangeBound: PostgresDataConvertible, Comparable {
```

Comment:
can we split this into `PostgresRangeEncodable` and `PostgresRangeDecodable`? Why does this inherit `Comparable`?

Reply:
I just pushed a commit where I limited the usage of `Comparable`: https://github.com/vapor/postgres-nio/pull/330/commits/3b2989981c31866306debab7eb6a87832ee02497. The reason it's needed at all is because in Swift's `RangeExpression` types, the `Bound` has to be `Comparable`: https://developer.apple.com/documentation/swift/rangeexpression/bound.

Reply:
I replaced `PostgresRangeBound` with `PostgresRangeEncodable` and `PostgresRangeDecodable` here: https://github.com/vapor/postgres-nio/pull/330/commits/9b8b3ec7a9de3031df8a84b54050d1317cfb61b4.

Reply:
Lovely!